### ISSUE TITLE
Filter serde exported variants in `wildcard_enum_match_arm`

### ISSUE THREAD

**Issue by mental32** [rust-lang/rust-clippy#5733]
Hi there! I frequently encounter `clippy::restriction` hits for `wildcard_enum_match_arm` but for items that are generated by `serde` and are private/hidden (notice leading underscore convention).

Could arms like this be selectively filtered out for this lint? I don't think it'll improve readability if I add this arm (I don't even think I can add this arm since the item is private) and I think these are closer to informal artifacts generated by `serde` rather than an api that should be relied on.

At the moment my go to solution is to `#[allow(...)]` it for the function its found in but it ends up reading like tedious noise.

```rust
error: wildcard match will miss any future added variants
   --> src/lib.rs:166:13
    |
166 |             _ => continue,
    |             ^ help: try this: `foo::_IMPL_DESERIALIZE_FOR_Foo::_serde::export::Ok(..)`
    |
note: the lint level is defined here
   --> src/lib.rs:5:22
    |
5   | #![deny(clippy::all, clippy::restriction)]
    |                      ^^^^^^^^^^^^^^^^^^^
    = note: `#[deny(clippy::wildcard_enum_match_arm)]` implied by `#[deny(clippy::restriction)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_enum_match_arm
```

Curious to hear thoughts on this!

**Comment by ThibsG** [comment]
Thanks for reporting this!
Can you provide minimal code example (not only backtrace) or at a least show portion of code that is triggering this ?

**Comment by mental32** [comment]
No problem: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=b4adc23a8806145a48f45bf35f184082

**Comment by flip1995** [comment]
This just needs a check for `in_external_macro`

@mental32 FYI: enabling the complete `restriction` group is not recommended, since it includes really _restrictive_ lints, that sometimes are in contrast with other lints or even go against idiomatic rust. These lints should only be enabled on a lint-by-lint basis and with careful consideration.

**Comment by mental32** [comment]
@flip1995 oh, nice to know! :sweat_smile: 

I tended to just switch all the groups on and filter away the stuff that annoyed me, It never occurred to me to work the other way.

**Comment by flip1995** [comment]
You can enable the whole `pedantic` group, if you don't mind sprinkling `allow`s throughout the code (Clippy itself does this). But IMO picking out lints from `pedantic` and `restriction` one by one is the better approach. This is not an official recommendation, just my personal preference, though.

If you want even more lints, you can also look at the `nursery` group, but these lints are buggy in one way or another, so be extra careful there.
